pkg/controller/certificates/controller.go

```diff
@@ -181,6 +187,51 @@ func (cc *CertificateController) maybeSignCertificate(key string) error {
 	}
 	csr := obj.(*certificates.CertificateSigningRequest)
 
+	func() {
+		if !cc.approveAllKubeletCSRs ||
```

Comment:
Ok, so my change results in approval leakage, so it's clearly not the right fix for the logic. By the way, why is this wrapped in a lambda?